features/consent/src/main/java/in/projecteka/jataayu/consent/viewmodel/GrantedConsentDetailsViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package `in`.projecteka.jataayu.consent.viewmodel
+
+import `in`.projecteka.jataayu.consent.Cache.ConsentDataProviderCacheManager
+import `in`.projecteka.jataayu.consent.repository.ConsentRepository
+import `in`.projecteka.jataayu.core.model.LinkedAccountsResponse
+import `in`.projecteka.jataayu.core.model.LinkedCareContext
+import `in`.projecteka.jataayu.core.model.Links
+import `in`.projecteka.jataayu.core.model.grantedconsent.GrantedConsentDetailsResponse
+import `in`.projecteka.jataayu.core.model.grantedconsent.LinkedHip
+import `in`.projecteka.jataayu.network.utils.PayloadLiveData
+import `in`.projecteka.jataayu.network.utils.fetch
+import `in`.projecteka.jataayu.presentation.callback.IDataBindingModel
+import `in`.projecteka.jataayu.util.extension.EMPTY
+import androidx.lifecycle.ViewModel
+
+class GrantedConsentDetailsViewModel(private val repository: ConsentRepository) : ViewModel() {
+
+    val linkedAccountsResponse = PayloadLiveData<LinkedAccountsResponse>()
+    val grantedConsentDetailsResponse = PayloadLiveData<List<GrantedConsentDetailsResponse>>()
+    internal var selectedProviderName = String.EMPTY
+
+    fun getLinkedAccounts() =
+        linkedAccountsResponse.fetch(repository.getLinkedAccounts())
+
+    fun getConsentRepository(): ConsentRepository = repository
+
+    fun getGrantedConsentDetails(requestId: String) {
+        grantedConsentDetailsResponse.fetch(repository.getGrantedConsentDetails(requestId))
+    }
+
+
+    fun getItems(grantedConsents: List<GrantedConsentDetailsResponse>, linkedAccounts: List<Links>?): Pair<List<IDataBindingModel>,Int> {
```

Comment:
write a better function for this. (yes, it can be done)

Reply:
try to make it event driven again. viewmodel should supply the objects via an event. view shouldn't request. 
a good example -> 
Tell Don't Ask - Martin Fowlermartinfowler.com